### PR TITLE
feat: [PL-43387]: Add support to migrate smp file secret

### PR DESCRIPTION
**Pull request by meenakshi-harness**
## Describe your changes
Add support in `fileMigration` api to migrate secrets in SMP en from CG to NG. This is case where encrypted file is stored in mongo.

Testing:
1. Migrated CG secret in NG through this API and tested secret decrypt in git connector using file secret.
2. Verified correct data migrated to config.files and config.chunks along with ngEncrytedRecords and secrets collection

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57016)

### CODE REVIEW

**Review by meenakshi-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57016)* status: 9 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 362 at r1](https://reviewable.io/reviews/harness/harness-core/57016#-NmB14rS2fyOsPvSi5Wy:-NmB5-r74Rfla3Lh0RG-:b3ift21) ([raw file](https://github.com/harness/harness-core/blob/f07b3e327ca0ddeec4a1ad782376b76923987553/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L362)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Do we need new argument encryptedFileContent? encryptedValue can be used to pass the file content.
</blockquote></details>

Yes but it can be confusing for caller as well as for us to have validation in case of SMP env

File: 100-migrator/src/main/java/io/harness/ngmigration/client/NGClient.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ Call<ResponseDTO<SecretResponseWrapper>> createSecretFileInternal(@Header(X_API_
       @Query(NGCommonEntityConstants.ORG_KEY) String orgIdentifier,
       @Query(NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier,
       @Query("encryptionKey") String encryptionKey, @Query("encryptedValue") String encryptedValue,
-      @Part("spec") RequestBody spec);
+      @Query("encryptedFileContent") String encryptedFileContent, @Part("spec") RequestBody spec);
```
Comment (lasdox) [inline 100-migrator/src/main/java/io/harness/ngmigration/client/NGClient.java:78]: Is it possible to have limitations for big file contents?

Comment (meenakshi-harness) [reply 100-migrator/src/main/java/io/harness/ngmigration/client/NGClient.java]: Removed from query param and moved to form data

**Comment by meenakshi-harness** [comment]
trigger uts

**Comment by meenakshi-harness** [comment]
trigger uts

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57016)* status: 9 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 362 at r1](https://reviewable.io/reviews/harness/harness-core/57016#-NmB14rS2fyOsPvSi5Wy:-NmFeCzqDqXEh6cYEKBc:b-drxgbo) ([raw file](https://github.com/harness/harness-core/blob/f07b3e327ca0ddeec4a1ad782376b76923987553/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L362)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Yes but it can be confusing for caller as well as for us to have validation in case of SMP env
</blockquote></details>

How does it relate to smp? The current logic of creating file secret also works for both SMP and SAAS. How does it matter here?

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 1 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57016)* status: 9 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 362 at r1](https://reviewable.io/reviews/harness/harness-core/57016#-NmB14rS2fyOsPvSi5Wy:-NmFqvaE9bYVNYQC3iDC:b13xlk2) ([raw file](https://github.com/harness/harness-core/blob/f07b3e327ca0ddeec4a1ad782376b76923987553/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L362)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

How does it relate to smp? The current logic of creating file secret also works for both SMP and SAAS. How does it matter here?
</blockquote></details>

As discussed since this is migration specific, we decided to go with extra parameter.

**Comment by meenakshi-harness** [comment]
trigger uts

**Review by lasdox — APPROVED** [review]

**Comment by meenakshi-harness** [comment]
trigger uts

**Comment by meenakshi-harness** [comment]
trigger ut9

**Comment by meenakshi-harness** [comment]
trigger messagecheck

**Review by lasdox — APPROVED** [review]

**Pull request merged**